src/inference/gibbs.jl

```diff
@@ -166,27 +166,45 @@ function AbstractMCMC.step!(
     # Iterate through each of the samplers.
     transitions = map(enumerate(spl.state.samplers)) do (i, local_spl)
         Turing.DEBUG && @debug "$(typeof(local_spl)) stepping..."
-
-        # Update the sampler's VarInfo.
+        
+        # Update the local sampler's VarInfo.
         local_spl.state.vi = spl.state.vi
 
         # Step through the local sampler.
         if transition === nothing
-            trans = AbstractMCMC.step!(rng, model, local_spl, N, nothing; kwargs...)
+            trans = gibbs_step!(rng, model, local_spl, N, nothing; kwargs...)
         else
-            trans = AbstractMCMC.step!(rng, model, local_spl, N, transition.transitions[i];
-                                       kwargs...)
+            trans = gibbs_step!(rng, model, local_spl, N, transition.transitions[i]; kwargs...)
         end
-
+        
         # After the step, update the master varinfo.
         spl.state.vi = local_spl.state.vi
 
-        trans
+        return trans
     end
 
     return GibbsTransition(spl, transitions)
 end
 
+
+"""
+    gibbs_step!(rng, model, spl, N, transition; kwargs...)
+
+Perform a conditional sampling step of the local sampler `spl` within a `Gibbs` sampler.  If the
+sampler is based on an `InferenceAlgorithm`, this just falls back to `AbstractMCMC.step!`.
+"""
+function gibbs_step!(
```

Comment:
I'm not clear on why we use `gibbs_step!` instead of just `step!`. What's the value add here?

Reply:
I honestely don't remember :) Presumably because `GibbsConditional` is not an `AbstractSampler`.  But removing it does not make it work not, so I did that.

Reply:
Perfect, thanks!

Reply:
It came up in https://github.com/TuringLang/Turing.jl/pull/1172#issuecomment-602189523. I don't remember exactly, but I think the idea was that you might want to perform some initial setup or configurations in the steps of the Gibbs sampler that you wouldn't perform in regular sampling. Currently that's all checked in the `step!` implementation (e.g., for HMC and ESS), but it would be cleaner to not mix these branches in the general `step!` implementation. The default implementation of `gibbs_step!` would just call `step!` (so not previous setup) but algorithms such as HMC could implement `gibbs_step!`, which would perform the setup and then call `step!`. Of course, `GibbsConditional` could then still just implement `step!` but I'm not sure if it makes sense to use GibbsConditional for anything else than Gibbs steps, so it might be a bit weird to keep it that general.

Reply:
Maybe it would actually make sense to keep `gibbs_step!`? :thinking:

Reply:
I feel like we should adhere to the interface as it currently is until we have a really good reason to deviate.

Reply:
Yes, maybe it's actually better to clean the Gibbs specific parts of `step!` in a separate PR.